pennylane/wires.py

```diff
@@ -155,9 +171,12 @@ def index(self, wire):
             if len(wire) != 1:
                 raise WireError("Can only retrieve index of a Wires object of length 1.")
 
-            return self.wire_tuple.index(wire.wire_tuple[0])
+            wire = wire.wire_tuple[0]
 
-        return self.wire_tuple.index(wire)
```

Comment:
Better to return a legible error here, this came up in discussion with Nathan I think.